HeatStroke/GOComponents/Components/ComponentPerspectiveCamera.cpp

```diff
@@ -0,0 +1,155 @@
+		glm::vec3(),
+		glm::vec3(),
+		glm::vec3(0.0f, 1.0f, 0.0f),
+		p_fFOV,
+		p_fAspectRatio,
+		p_fNearClip,
+		p_fFarClip
+	),
+	m_pTargetGameObject(p_pTargetGameObject)
+{
+	SceneManager::Instance()->AddPerspectiveCamera(&m_mScenePerspectiveCamera);
+}
+
+HeatStroke::ComponentPerspectiveCamera::~ComponentPerspectiveCamera()
+{
+	SceneManager::Instance()->RemovePerspectiveCamera(&m_mScenePerspectiveCamera);
+}
+
+HeatStroke::Component* HeatStroke::ComponentPerspectiveCamera::CreateComponent(
+	HeatStroke::GameObject* p_pGameObject, 
+	tinyxml2::XMLNode* p_pBaseNode,
+	tinyxml2::XMLNode* p_pOverrideNode)
+{
+	assert(p_pGameObject != nullptr);
+
+	// Defaults
+	float fFOV = 45.0f;
+	float fAspectRatioWidth = 1280.0f;
+	float fAspectRatioHeight = 720.0f;
+	float fNearClip = 0.1f;
+	float fFarClip = 1000.0f;
+	std::string strTargetGUID("");
+	GameObject* pTargetGameObject = nullptr;
+
+	if (p_pBaseNode != nullptr)
+	{
+		ParseNode(p_pBaseNode, fFOV, fAspectRatioWidth, fAspectRatioHeight, fNearClip, fFarClip, strTargetGUID);
+	}
+	if (p_pOverrideNode != nullptr)
+	{
+		ParseNode(p_pOverrideNode, fFOV, fAspectRatioWidth, fAspectRatioHeight, fNearClip, fFarClip, strTargetGUID);
+	}
+
+	// Apply the target if one was set.
+	if (strTargetGUID != "")
+	{
+		pTargetGameObject = p_pGameObject->GetManager()->GetGameObject(strTargetGUID);
+	}
+
+	return new ComponentPerspectiveCamera(
+		p_pGameObject, 
+		fFOV, 
+		fAspectRatioWidth / fAspectRatioHeight, 
+		fNearClip, 
+		fFarClip,
+		pTargetGameObject
+	);
+}
+
+void HeatStroke::ComponentPerspectiveCamera::Update(const float p_fDelta)
+{
+	glm::vec3 vPosition = m_pGameObject->GetTransform().GetTranslation();
+	m_mScenePerspectiveCamera.SetPosition(vPosition);
+
+	// If this is not nullptr, then we have a target we're following.
+	if (m_pTargetGameObject != nullptr)
+	{
+		m_mScenePerspectiveCamera.SetTarget(m_pTargetGameObject->GetTransform().GetTranslation());
+	}
+	// Otherwise, just face in the same direction as the GameObject.
+	else
+	{
+		glm::vec3 vForward = glm::normalize(m_pGameObject->GetTransform().GetRotation() * glm::vec3(0.0f, 0.0f, 1.0f));
+		m_mScenePerspectiveCamera.SetTarget(vPosition + vForward);
+	}
+}
+
+
+void HeatStroke::ComponentPerspectiveCamera::SetScenePerspectiveCamera(const HeatStroke::ScenePerspectiveCamera& p_mScenePerspectiveCamera)
+{
+	SceneManager::Instance()->RemovePerspectiveCamera(&m_mScenePerspectiveCamera);
+	m_mScenePerspectiveCamera = p_mScenePerspectiveCamera;
+	SceneManager::Instance()->AddPerspectiveCamera(&m_mScenePerspectiveCamera);
+}
+
+
+void HeatStroke::ComponentPerspectiveCamera::ParseNode(
+	tinyxml2::XMLNode* p_pNode,
+	float& p_fFOV,
+	float& p_fAspectRatioWidth,
+	float& p_fAspectRatioHeight,
+	float& p_fNearClip,
+	float& p_fFarClip,
+	std::string& p_strTargetGUID
+)
+{
+	assert(p_pNode != nullptr);
+	assert(strcmp(p_pNode->Value(), "GOC_PerspectiveCamera") == 0);
+
+	for (tinyxml2::XMLNode*  pChildNode = p_pNode->FirstChild();
```

Comment:
Same as before: is there a reason for `FirstChild()` over `FirstChildElement()`?